packages/rum/src/domain/record/privacy.ts

```diff
@@ -1,91 +1,407 @@
+      }
+      const autocomplete = inputElement.getAttribute('autocomplete')
+      if (autocomplete && autocomplete.indexOf('cc-') === 0) {
+        return NodePrivacyLevelInternal.MASK
+      }
+    }
+
+    // Customers should first specify privacy tags using HTML attributes
+    switch (privAttr) {
+      case PRIVACY_ATTR_VALUE_ALLOW:
+        return NodePrivacyLevelInternal.ALLOW
+      case PRIVACY_ATTR_VALUE_MASK:
+        return NodePrivacyLevelInternal.MASK
+      case PRIVACY_ATTR_VALUE_MASK_FORMS_ONLY:
+      case PRIVACY_ATTR_VALUE_INPUT_IGNORED: // Deprecated, now aliased
+      case PRIVACY_ATTR_VALUE_INPUT_MASKED: // Deprecated, now aliased
+        return NodePrivacyLevelInternal.MASK_FORMS_ONLY
+      case PRIVACY_ATTR_VALUE_HIDDEN:
+        return NodePrivacyLevelInternal.HIDDEN
+      case PRIVACY_ATTR_VALUE_MASK_SEALED:
+        return NodePrivacyLevelInternal.MASK_SEALED
+      case PRIVACY_ATTR_VALUE_MASK_FORMS_ONLY_SEALED:
+        return NodePrivacyLevelInternal.MASK_FORMS_ONLY_SEALED
+    }
+
+    // But we also need to support class based privacy tagging for certain frameworks
+    if (elNode.classList.contains(PRIVACY_CLASS_ALLOW)) {
+      return NodePrivacyLevelInternal.ALLOW
+    } else if (elNode.classList.contains(PRIVACY_CLASS_MASK)) {
+      return NodePrivacyLevelInternal.MASK
+    } else if (elNode.classList.contains(PRIVACY_CLASS_HIDDEN)) {
+      return NodePrivacyLevelInternal.HIDDEN
+    } else if (
+      elNode.classList.contains(PRIVACY_CLASS_MASK_FORMS_ONLY) ||
+      elNode.classList.contains(PRIVACY_CLASS_INPUT_MASKED) || // Deprecated, now aliased
+      elNode.classList.contains(PRIVACY_CLASS_INPUT_IGNORED) // Deprecated, now aliased
+    ) {
+      return NodePrivacyLevelInternal.MASK_FORMS_ONLY
+    } else if (shouldIgnoreElement(elNode)) {
+      // such as for scripts
+      return NodePrivacyLevelInternal.IGNORE
+    }
+  } else if (node.nodeType === Node.DOCUMENT_NODE) {
+    return getInitialPrivacyLevel()
+  }
+
+  // Other node types cannot be tagged directly
+  return NodePrivacyLevelInternal.NOT_SET
+}
+
+export function getAttributesForPrivacyLevel(
+  element: Element,
+  nodePrivacyLevel: NodePrivacyLevel
+): Record<string, string | number | boolean> {
+  /*
+    NEVER ALLOW	`value` under these conditions
+    input[type=password]	
+    input[autocomplete=cc-number/cc-csc/cc-exp/cc-exp-month/cc-exp-year]	
+    MAYBE: input[type=tel,email]	
+  */
+  if (nodePrivacyLevel === NodePrivacyLevel.HIDDEN) {
+    return {}
+  }
+
+  const tagName = element.tagName
+  const safeAttrs: Record<string, string> = {}
+
+  getAttributeEntries(element.attributes).forEach(([attrName, attrVal]) => {
+    // Never take those attributes into account, as they will be conditionally added in `serializeElementNode`
+    if (attrName !== 'value' && attrName !== 'selected' && attrName !== 'checked') {
+      safeAttrs[attrName] = attrVal
+    }
+  })
+
+  if (nodePrivacyLevel === NodePrivacyLevel.MASK) {
```

Comment:
Could you clarify what you mean here?

Reply:
Instead of overriding attribute values or looping on attributes twice, you could do something like:

```ts
getAttributeEntries(element.attributes).forEach(([attrName, attrVal]) => {
    if (nodePrivacyLevel === NodePrivacyLevel.MASK && attrVal &&
     (attrName === 'title' || attrName === 'alt' || ... || attrName.indexOf('data-') === 0 && attrName !== PRIVACY_ATTR_NAME)) {
      attrVal = CENSORED_STRING_MASK
    }
   // you may have more special cases for input value/selected/checked here
    safeAttrs[attrName] = attrVal
  })
```

Or even have a `getAttributeForPrivacyLevel` that you can reuse when you are processing a single attribute, ex:

```ts
function getAttributesForLevel(...) {
...
for (let i = 0; i < element.attributes.length; i += 1) {
    let attrName = element.attributes.item(i).name
    safeAttrs[attrName] = getAttributeForPrivacyLevel(element, attrName, nodePrivacyLevel)
}
...
}

function getAttributeForPrivacyLevel(element, attrName, nodePrivacyLevel) {
   if (nodePrivacyLevel === NodePrivacyLevel.MASK && attrVal &&
     (attrName === 'title' || attrName === 'alt' || ... || attrName.indexOf('data-') === 0 && attrName !== PRIVACY_ATTR_NAME)) {
      return CENSORED_STRING_MASK
    }
   // you may have more special cases for input value/selected/checked here
    return element.getAttribute(attrName)
}

```